packages/flutter/test/widgets/editable_text_test.dart

```diff
@@ -161,7 +160,7 @@ void main() {
     controller.text = 'test';
     await tester.idle();
     expect(tester.testTextInput.editingState['text'], equals('test'));
-    expect(tester.testTextInput.setClientArgs['inputType'], equals('TextInputType.text'));
+    expect(tester.testTextInput.setClientArgs['inputType']['type'], equals('TextInputType.text'));
     expect(tester.testTextInput.setClientArgs['inputAction'], equals('TextInputAction.done'));
   });
 
```

Comment:
add tests for:
 - `==` and `hashCode` (not the specific code, just that it's the same when == is the same, and different otherwise)
 - toString
 - that signed and decimal can both be set, that they default to false for number and null for non-number

Reply:
Added in text_field.test.dart.